integration/users/usersinfo_test.go

```diff
@@ -0,0 +1,496 @@
+			dbSuffix: "_example",
+			payload: bson.D{
+				{"usersInfo", bson.A{
+					bson.D{{"user", "b"}, {"db", "TestUsersinfo_example"}},
+					bson.D{{"user", "c"}, {"db", "TestUsersinfo_example"}},
+				}},
+			},
+			expected: bson.D{
+				{"users", bson.A{
+					bson.D{
+						{"_id", "TestUsersinfo_example.b"},
+						{"user", "b"},
+						{"db", "TestUsersinfo_example"},
+						{"roles", bson.A{}},
+					},
+					bson.D{
+						{"_id", "TestUsersinfo_example.c"},
+						{"user", "c"},
+						{"db", "TestUsersinfo_example"},
+						{"roles", bson.A{}},
+					},
+				}},
+				{"ok", float64(1)},
+			},
+		},
+		"MissingDBFieldName": {
+			dbSuffix: "_example",
+			payload: bson.D{
+				{"usersInfo", bson.A{
+					bson.D{{"user", "missing_db"}},
+				}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Message: "UserName must contain a field named: db",
+				Name:    "BadValue",
+			},
+		},
+		"MissingUserFieldName": {
+			dbSuffix: "_example",
+			payload: bson.D{
+				{"usersInfo", bson.A{
+					bson.D{{"db", "TestUsersinfo_example"}},
+				}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Message: "UserName must contain a field named: user",
+				Name:    "BadValue",
+			},
+		},
+		"ForAllDBs": {
+			dbSuffix: "_example",
+			payload: bson.D{{
+				"usersInfo", bson.D{
+					{"forAllDBs", true},
+				},
+			}},
+			hasUser: map[string]struct{}{
+				"TestUsersinfo.one":                {},
+				"TestUsersinfo.two":                {},
+				"TestUsersinfo_example.a":          {},
+				"TestUsersinfo_example.b":          {},
+				"TestUsersinfo_example.c":          {},
+				"TestUsersinfo_few.i":              {},
+				"TestUsersinfo_few.j":              {},
+				"TestUsersinfo_another.singleuser": {},
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var res bson.D
+			dbName := dbPrefix + tc.dbSuffix
+			err := client.Database(dbName).RunCommand(ctx, tc.payload).Decode(&res)
+			if tc.err != nil {
+				integration.AssertEqualAltCommandError(t, *tc.err, tc.altMessage, err)
+				return
+			}
+
+			require.NoError(t, err)
+
+			actual := integration.ConvertDocument(t, res)
+			actual.Remove("$clusterTime")
+			actual.Remove("operationTime")
+
+			actualUsers := must.NotFail(actual.Get("users")).(*types.Array)
+
+			var (
+				expected      *types.Document
+				expectedUsers *types.Array
+			)
+
+			if tc.expected != nil {
+				expected = integration.ConvertDocument(t, tc.expected)
+				expectedUsers = must.NotFail(expected.Get("users")).(*types.Array)
+				require.Equal(t, expectedUsers.Len(), actualUsers.Len(), "users length")
+			}
+
+			foundUsers := map[string]struct{}{}
+			for i := 0; i < actualUsers.Len(); i++ {
+				au, err := actualUsers.Get(i)
+				require.NoError(t, err)
+				actualUser := au.(*types.Document)
+
+				assert.False(t, tc.hasUser == nil && tc.expected == nil, "set at least one expectation for %q", name)
```

Comment:
```suggestion
				require.False(t, tc.hasUser == nil && tc.expected == nil, "set at least one expectation for %q", name)
```

There is no point in continuing with a broken test case.

But that also does not test the case when both _are_ set. So I guess it should be `require.True((tc.hasUser == nil) != (tc.expected == nil))` or something